runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/StreamingDataflowWorker.java

```diff
@@ -569,7 +599,10 @@ private static GrpcWindmillStreamFactory createWindmillStreamFactory(
         .setMaxBackOffSupplier(() -> maxBackoff)
         .setLogEveryNStreamFailures(options.getWindmillServiceStreamingLogEveryNStreamFailures())
         .setStreamingRpcBatchLimit(options.getWindmillServiceStreamingRpcBatchLimit())
-        .build();
+        .setSendKeyedGetDataRequests(
+            !options.isEnableStreamingEngine()
+                || !DataflowRunner.hasExperiment(
+                    options, "streaming_engine_send_new_heartbeat_requests"));
```

Comment:
why not set the health check here?

Reply:
synced offline

Reply:
nit: nice to update PR with discussed reason when responding, for other readers

Reply:
sgtm sorry about that